File: TestSpicerExample.rb
load "TestSpicerAPI.rb"

sessionId = start_test('CucumberRuby', 'Testing', 'example', 'This is an example', 'Something', 'None')

name = get_random_name(sessionId)['firstname']
lastname = get_random_name(sessionId)['lastname']
email = get_random_email(sessionId)['email']
age = get_random_age(sessionId, "28", "35" )['year']
areacode = get_random_phone(sessionId)['areacode']
phone = get_random_phone(sessionId)['number']
string = get_random_alphabet_string(sessionId, "10")['string']

postalcode = get_random_alphabet_string(sessionId, "5")['string']

puts name
puts lastname
puts email.downcase
puts age
puts areacode << " " << phone
puts string
puts postalcode

finish_test(sessionId)

File: testSpicerExample.py
import TestSpicerAPI

testApi = TestSpicerAPI.TestSpicerAPI()

sessionId = testApi.startTest('Project', 'Suite', 'Name', 'Description', 'Functional Area', 'Risk Coverage')

age = testApi.getRandomAge(sessionId,1,13)['year']
areacode = testApi.getRandomTelephone(sessionId)['areacode']
number = testApi.getRandomTelephone(sessionId)['number']
completePhone = areacode+' '+number
email = testApi.getRandomEmail(sessionId)['email']
name = testApi.getRandomName(sessionId)['firstname']
lastname = testApi.getRandomName(sessionId)['lastname']
password = testApi.getRandomAlphaNumeric(sessionId, 10)['string']
zipcode = testApi.getRandomUSAZipCode(sessionId, 10000, 99999)['number']
#there's a bug on Test Spice this should be 0,3 to generate 1 and 2 values but is actually needed to go up by one, reported it to them
day = testApi.getRandomDate(sessionId,'01011900','01011995')['day']
month = testApi.getRandomDate(sessionId,'01011900','01011995')['month']
year = testApi.getRandomDate(sessionId,'01011900','01011995')['year']

print ' '
print 'data Obtained'
print ' '
print 'Name: ' + name
print 'Lastname: ' + lastname
print 'Email: ' + email
print 'Password: ' + password
print 'Zip Code: ' + zipcode
print 'Birthdate'
print 'day: ' + day + ' month: ' + month + ' year: ' + year
print "Random Age: "+str(age)

finishTest = testApi.finishTest(sessionId)